### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by trisbsamson**
Pull request for initial submission of COMP3710 Final Report contribution to PatternFlow.

Provided implementation solves the ISIC2018 semantic segmentation problem with the standard U-Net architecture. Achieves an average Dice coefficient on the test set of ~0.82. 

Notes:
 - Usage details and dependencies available in README.md

### CODE REVIEW

**Comment by trisbsamson** [comment]
Additional details if required:
Student Number: 43971198

**Comment by SiyuLiu0329** [comment]
- Sufficiently detailed commit messages
- Code valid for Easy Difficulty
- Good results

Other Feedback
- (OPTIONAL) Would put main logic in `if __name__ == '__main__'` for the sake of following python convention.
- it is more technically correct to use 1 output class and sigmoid activation, though nothing wrong with the current implementation

**Comment by rexsw** [comment]
Code looks good for easy difficulty

For next time;

Tensorflow has functions for image opening and more so there’s no need to also use PIL in your project. Same applies with numpy and shuffling data.

Comments such as 
     # shuffle the data array
    np.random.shuffle(data)
Add nothing to the readability of the code and should be avoided. Overall comments were well done

File: recognition/TBS-UNet_ISIC/driver.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
+import os
```
Comment (shakes76) [inline recognition/TBS-UNet_ISIC/driver.py:1]: Code header block should be added for all scripts/source files

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme OK
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Request Description. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.

**Pull request merged**